shared-scripts.sh

```diff
@@ -125,6 +125,7 @@ function _verifyAPIExtract {
     # download [repo, .cache from s3]
     loadCache repo $CODEBUILD_SRC_DIR
     loadCache .cache $HOME/.cache
+    unset IS_AMPLIFY_CI
```

Comment:
Why is this included in this PR?

Reply:
short story: codebuild sets IS_AMPLIFY_CI globally, but circleci doesn't, so this achieves parity.
long story:
this PR changed the API slightly, which shined a light on something we didn't see before in codebuild, but happens all the time in circleci. That is, verify-api-extract actually downloads the amplify-cli & installs it due to a post install script for the amplify-cli-npm package. Downloading the CLI has no effect on the API step, but when because its running in codebuild CI - the IS_AMPLIFY_CI flag makes it try to download from s3 (pkg cli) using the binary hash. CircleCI does not set IS_AMPLIFY_CI for the verify-api-extract, so it succeeds there because it just downloads the production pkg cli. Download the pkg cli has no affect on verify-api-extract's result.

Reply:
Would you mind doing a quick one-off PR and merging it first? It will improve discoverability for the change and then this comment can become the PR description.

Reply:
https://github.com/aws-amplify/amplify-cli/pull/13019